configure_metrics_agent.sh

```diff
@@ -10,18 +10,65 @@ BIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 INSTALL_DIR=$( cd "$BIN_DIR/.." && pwd )
 MON_SUDOERS_FILE="/etc/sudoers.d/mon-agent"
 
+MON_AGENT_DIR="/etc/monasca/agent"
+MON_SYSTEMD_DIR="/etc/systemd/system"
+
 if [ ! -e "${MON_SUDOERS_FILE}" ]; then
     echo "mon-agent ALL=(ALL) NOPASSWD:ALL" | sudo tee "${MON_SUDOERS_FILE}" >> /dev/null
 fi
 
 sudo mkdir -p /etc/monasca
 
-sudo "${BIN_DIR}/python" "${BIN_DIR}/monasca-setup" "$@"
+# Check if file exist, if yes and OVERWRITE_CONF set to false, create backup
+# in same folder
+function protect_overwrite() {
+    local protected_file="$1"
+    if [ -f "${protected_file}" ]; then
+        if [ "${OVERWRITE_CONF}" = "false" ]; then
+            warn "${protected_file} already exists"
+            warn "If you want to overwrite it you need to use '--overwrite_conf'"
+            \cp -f "${protected_file}" "${protected_file}.backup"
+            return
+        else
+            inf "Existing ${protected_file} file will be overwritten"
+        fi
+    fi
+}
+
+# Restore file to original location from backup copy if OVERWRITE_CONF is false
+function protect_restore() {
+    local protected_file="$1"
+    if [ "${OVERWRITE_CONF}" = "false" ]; then
+        warn "Restoring original ${protected_file}"
+        warn "If you want to overwrite it you need to use '--overwrite_conf'"
+        \mv -f "${protected_file}.backup" "${protected_file}"
+        return
+    fi
+}
+
+function run_monasca_setup() {
+    local all_args="$1"
+
+    # All this files will be uncoditionaly overwritten by monasca-setup
+    # so we are creating they backups if OVERWRITE_CONF is set to false
+    protect_overwrite "${MON_AGENT_DIR}/agent.yaml"
```

Comment:
I think it's better to call `protect_overwrite` if `protected_file` exists and `OVERWRITE_CONF` is `false`, and define these 3 files as `list` and use for loop.

Reply:
Done